simulator/func_sim/t/unit_test.cpp

```diff
@@ -95,6 +95,17 @@ TEST_CASE( "Run one instruction: Func_Sim")
     CHECK( FuncSimAndMemory<MIPS32>().run_trace( smc_code, 1) == Trap::NO_TRAP);
 }
 
+TEST_CASE( "FuncSim: Register R/W")
+{
+    auto simMIPS32 = Simulator::create_simulator( "mips32", true, false);
```

Comment:
I suggest to use FuncSim<> ctors directly. There are no tests for factory methods yet.